lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -1233,6 +1245,67 @@ void OpaqueStorageAllocation::allocateValue(SILValue value) {
   createStackAllocationStorage(value);
 }
 
+/// Enum and existential projections both perform initialization side-effects.
+/// Consequently they can only be materialized once on any path.
+///
+/// If the user is not a phi (`allowsInit` == true), then it is always possible
+/// to materialize initialization at the single point at which the address must
+/// be available.
+///
+/// Materializing an address for a coalesced phi (`allowsInit` == false),
+/// however, cannot rematerialize initialization, because that would require
+/// the address to be available on both sides of the phi.  But we can't have an
+/// address phi.
+///
+/// Concretely, given:
+///
+///     left:
+///       %e1 = init_existential_value %v1
+///       br merge(%e1 : $P)
+///     right:
+///       %e2 = init_existential_value %v2
+///       br merge(%e2 : $P)
+///     merge(%e : @owned $P):
+///
+/// we can't produce a single init_existential_addr instruction in the `merge`
+/// block
+///
+///     merge:
+///       init_existential_addr ???
+///
+/// because doing so would require an address phi
+///
+///     merge(%addr : $*): // invalid!
+///       init_existential_addr %addr
+///
+/// As a result of the above consideration, when this function is called for a
+/// phi, it returns a null value if the storage involves either sort of
+/// effectful initialization.
+SILValue AddressLoweringState::getEarliestStoragePoint(SILValue userValue,
```

Comment:
Can we be explicit about the relationship between this function and  getProjectionInsertionPoint which is called later. Like, if this returns a valid address, the getProjectionInsertionPoint must be guaranteed to find a dominating address, so all dependencies discovered here need to be rediscovered there.

Reply:
Fixed.